client/api/filesystem/v1alpha1/api.proto

```diff
@@ -2,27 +2,156 @@ syntax = "proto3";
-// here as an example of what API group and versions will look like. It will actually
-// be implemented in a later patch.
-
 service Filesystem {
-    // PathExists checks if the given path exists on the host.
+    // PathExists checks if the requested path exists in the host's filesystem
     rpc PathExists(PathExistsRequest) returns (PathExistsResponse) {}
+
+    // Mkdir creates a directory at the requested path in the host's filesystem
+    rpc Mkdir(MkdirRequest) returns (MkdirResponse) {}
+
+    // Rmdir removes the directory at the requested path in the host's filesystem.
+    // This may be used for unlinking a symlink created through LinkPath
+    rpc Rmdir(RmdirRequest) returns (RmdirResponse) {}
+
+    // LinkPath creates a local directory symbolic link between a source path 
+    // and target path in the host's filesystem
+    rpc LinkPath(LinkPathRequest) returns (LinkPathResponse) {}
+}
+
+// Context of the paths used for path prefix validation
+enum PathContext {
+    // Indicates the kubelet-csi-plugins-path parameter of csi-proxy be used as the path context
+    PLUGIN = 0;
+    // Indicates the kubelet-pod-path parameter of csi-proxy be used as the path context
+    CONTAINER = 1;
 }
 
 message PathExistsRequest {
-    // The path to check in the host file system.
+    // The path whose existence we want to check in the host's filesystem
     string path = 1;
+
+    // Context of the path parameter.
+    // This is used to determine the root for relative path parameters
+    PathContext context = 2;
 }
 
 message PathExistsResponse {
-    bool success = 1;
+    // Error message if any. Empty string indicates success
+    string error = 1;
+
+    // Indicates whether the path in PathExistsRequest exists in the host's filesystem
+    bool exists = 2;
+}
+
+message MkdirRequest {
+    // The path to create in the host's filesystem.
+    // All special characters allowed by Windows in path names will be allowed
+    // except for restrictions noted below. For details, please check:
+    // https://docs.microsoft.com/en-us/windows/win32/fileio/naming-a-file
+    // Non-existent parent directories in the path will be automatically created.
+    // Directories will be created with Read and Write privileges of the Windows
+    // User account under which csi-proxy is started (typically LocalSystem).
+    //
+    // Restrictions:
+    // If an absolute path (indicated by a drive letter prefix: e.g. "C:\") is passed,
+    // depending on the context parameter of this function, the path prefix needs 
+    // to match the paths specified either as kubelet-csi-plugins-path
+    // or as kubelet-pod-path parameters of csi-proxy.
+    // If a relative path is passed, depending on the context parameter of this 
+    // function, the path will be considered relative to the path specified either as 
+    // kubelet-csi-plugins-path or as kubelet-pod-path parameters of csi-proxy.
```

Comment:
In what case kubelet-csi-plugins-path will be used? and what case for kubelet-pod-path?

Reply:
I will add that in the comments. For `NodeStageVolume` calls, kubelet-csi-plugins-path is expected to be used to create `C:\var\lib\kubelet\plugins\kubernetes.io\csi\pv\<pv-name>\globalmount`. For `NodePublishVolume` calls,, kubelet-pod-path is expected to be used to create `C:\var\lib\kubelet\pods\<pod-uuid>\volumes\kubernetes.io~csi\<pvc-name>\mount`.